iidm/iidm-impl/src/test/java/com/powsybl/iidm/network/impl/BranchTest.java

```diff
@@ -0,0 +1,91 @@
+        Assertions.assertNull(branch1.getNullableCurrentLimits2());
+
+        Assertions.assertNotNull(branch1.getActivePowerLimits(TwoSides.ONE));
+        Assertions.assertNotNull(branch1.getActivePowerLimits(TwoSides.TWO));
+        Assertions.assertNotNull(branch1.getNullableActivePowerLimits(TwoSides.ONE));
+        Assertions.assertNotNull(branch1.getNullableActivePowerLimits(TwoSides.TWO));
+        Assertions.assertNotNull(branch1.getApparentPowerLimits(TwoSides.ONE));
+        Assertions.assertNotNull(branch1.getApparentPowerLimits(TwoSides.TWO));
+        Assertions.assertNotNull(branch1.getNullableApparentPowerLimits(TwoSides.ONE));
+        Assertions.assertNotNull(branch1.getNullableApparentPowerLimits(TwoSides.TWO));
+
+        Assertions.assertNotNull(branch1.getNullableLimits(LimitType.ACTIVE_POWER, TwoSides.ONE));
+        Assertions.assertNotNull(branch1.getNullableLimits(LimitType.ACTIVE_POWER, TwoSides.TWO));
+        Assertions.assertNotNull(branch1.getNullableLimits(LimitType.APPARENT_POWER, TwoSides.ONE));
+        Assertions.assertNotNull(branch1.getNullableLimits(LimitType.APPARENT_POWER, TwoSides.TWO));
+    }
+
+    @Test
+    void testCurrentLimitsBranches() {
+        // current limits
+        Network network2 = EurostagTutorialExample1Factory.createWithFixedCurrentLimits();
+        Branch<?> branch2 = network2.getBranch("NHV1_NHV2_2");
+        Assertions.assertNull(branch2.getNullableActivePowerLimits1());
+        Assertions.assertNull(branch2.getNullableActivePowerLimits2());
+        Assertions.assertNull(branch2.getNullableApparentPowerLimits1());
+        Assertions.assertNull(branch2.getNullableApparentPowerLimits2());
+
+        Assertions.assertNotNull(branch2.getNullableCurrentLimits1());
+        Assertions.assertEquals(LimitType.CURRENT, branch2.getNullableCurrentLimits1().getLimitType());
+        Assertions.assertEquals(1100, branch2.getNullableCurrentLimits1().getPermanentLimit());
+        Assertions.assertEquals(2, branch2.getNullableCurrentLimits1().getTemporaryLimits().size());
+
+        Assertions.assertNotNull(branch2.getNullableCurrentLimits2());
+        Assertions.assertEquals(LimitType.CURRENT, branch2.getNullableCurrentLimits2().getLimitType());
+        Assertions.assertEquals(500, branch2.getNullableCurrentLimits2().getPermanentLimit());
+        Assertions.assertEquals(0, branch2.getNullableCurrentLimits2().getTemporaryLimits().size());
+
+        Assertions.assertNotNull(branch2.getCurrentLimits(TwoSides.ONE));
+        Assertions.assertNotNull(branch2.getCurrentLimits(TwoSides.TWO));
+        Assertions.assertNotNull(branch2.getNullableCurrentLimits(TwoSides.ONE));
+        Assertions.assertNotNull(branch2.getNullableCurrentLimits(TwoSides.TWO));
+
+        Assertions.assertNotNull(branch2.getNullableLimits(LimitType.CURRENT, TwoSides.ONE));
+        Assertions.assertNotNull(branch2.getNullableLimits(LimitType.CURRENT, TwoSides.TWO));
```

Comment:
Could you also add a test on the permanent limit value for each non-null item.
In the current state, nothing verify that the returned values are the expected one (for instance, side 1 and side 2 could be mixed here).

Reply:
ok